testdata/grammar_outside/elmamun.gap

```diff
@@ -0,0 +1,203 @@
+  answer heinz(answer, Rope, answer);
+  answer nil(void);
+  choice [answer] h([answer]);
+}
+
+algebra alg_enum auto enum;
+algebra alg_count auto count;
+
+algebra alg_pretty implements sig_elmamun(alphabet=char, answer=Rope) {
+  Rope number(int value) {
+    Rope res;
+    append(res, value);
+    return res;
+  }
+  Rope add(Rope left, char opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope heinz(Rope left, Rope opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope mult(Rope left, char opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope minus(Rope left, char opSymbol, Rope right) {
+    Rope res;
+    append(res, '(');
+    append(res, left);
+    append(res, opSymbol);
+    append(res, right);
+    append(res, ')');
+    return res;
+  }
+  Rope nil(void) {
+    Rope res;
+    return res;
+  }
+  choice [Rope] h([Rope] candidates) {
+    return candidates;
+  }
+}
+
+algebra alg_buyer implements sig_elmamun(alphabet=char, answer=int) {
+  int number(int value) {
+    return value;
+  }
+  int add(int left, char opSymbol, int right) {
+    return left + right;
+  }
+  int heinz(int left, Rope opSymbol, int right) {
+    return left + right;
+  }
+  int mult(int left, char opSymbol, int right) {
+    return left * right;
+  }
+  int minus(int left, char opSymbol, int right) {
+    return left - right;
+  }
+  int nil(void) {
+    return 0;
+  }
+  choice [int] h([int] candidates) {
+    return list(minimum(candidates));
+  }
+}
+algebra alg_seller extends alg_buyer {
+  choice [int] h([int] candidates) {
+    return list(maximum(candidates));
+  }
+}
+
+algebra alg_time implements sig_elmamun(alphabet=char, answer=int) {
+  int number(int value) {
+    return 0;
+  }
+  int add(int left, char opSymbol, int right) {
+    if (left > right) {
+      return left + 2;
+    } else {
+      return right + 2;
+    }
+  }
+  int heinz(int left, Rope opSymbol, int right) {
+    if (left > right) {
+      return left + 2;
+    } else {
+      return right + 2;
+    }
+  }
+  int mult(int left, char opSymbol, int right) {
+    if (left > right) {
+      return left + 5;
+    } else {
+      return right + 5;
+    }
+  }
+  int minus(int left, char opSymbol, int right) {
+    if (left > right) {
+      return left + 3;
+    } else {
+      return right + 3;
+    }
+  }
+  int nil(void) {
+    return 0;
+  }
+  choice [int] h([int] candidates) {
+    return list(minimum(candidates));
+  }
+}
+
+grammar gra_elmamun uses sig_elmamun(axiom = formula) {
+  formula = number(INT)
+	  | add(formula, CHAR, formula)
```

Comment:
did you forget the '+' here?

Reply:
not really. This is a test case and no productive code. Here, I wanted to see why the terminal parser CHAR could be specialized to '*' but ROPE was not able to only parse "manfred", see https://github.com/jlab/gapc/blob/63bd6bf6dc5b220c613e874fd0e9f7ebe3ff3fe9/testdata/regresstest/config#L586-L587

I hope nobody erroneously uses this file for "productive" purposes. Please refer to https://github.com/jlab/ADP_collection/blob/master/elmamun.gap instead!